src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -527,7 +527,7 @@ private enum NamespaceParts
                                 // incomplete members must be processed before we add any nodes to the body:
                                 ReduceIncompleteMembers(ref pendingIncompleteMembers, ref openBraceOrSemicolon, ref body, ref initialBadNodes);
 
-                                var attribute = this.ParseAttributeDeclaration();
+                                var attribute = this.ParseAttributeDeclaration(inExpressionContext: parentKind == SyntaxKind.CompilationUnit);
```

Comment:
`[` at the top level could start an attribute, or could start a collection literal.   However, `[` inside a namespace always starts an attribute.  Good test cases @jnm2  :)

Reply:
while not legal, the parser should also accept:

```
[IAmAnAttribute(1, 2, 3)]
[on, a, collection, literal].DoWork();
```